github/src/main/java/com/redhat/ipaas/github/GitHubServiceImpl.java

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.redhat.ipaas.github;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.PostConstruct;
+
+import com.redhat.ipaas.github.extended.ExtendedContentsService;
+import org.eclipse.egit.github.core.Repository;
+import org.eclipse.egit.github.core.RepositoryContents;
+import org.eclipse.egit.github.core.client.GitHubClient;
+import org.eclipse.egit.github.core.service.RepositoryService;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Service;
+
+/**
+ * @author roland
+ * @since 08/03/2017
+ */
+@Service
+@ConditionalOnProperty(value = "github.enabled", matchIfMissing = true, havingValue = "true")
+public class GitHubServiceImpl implements GitHubService {
+
+    private RepositoryService repoService;
+    private ExtendedContentsService contentsService;
+
+    @Value("github.service")
+    private String gitHubHost;
+
+    @PostConstruct
+    public void init() {
+        GitHubClient client = new GitHubClient(gitHubHost != null ? gitHubHost : "ipaas-github-proxy");
```

Comment:
Would it make sense to put the token into a ThreadLocal via a filter ? Otherwise it really clutters the API as it then has to be extracted individually in each handler and then passed to the GitHub Service for each call.

We could add a `AuthTokenHolder` class to the shared packages which is accessible from the other services. If we want to continue with the proxy approach, guess that's an easy way to make the token available globally.

Any thought on this ? Since its within a filter you also can remove the ThreadLocal directly in the same code where you have created it in a finally block, so there is no danger of leaking.